app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        key = (args, tuple(sorted(kwargs.items())))
```

Comment:
Using `args` directly as part of the dictionary key can lead to runtime errors if any of its elements are mutable. Consider converting `args` to a fully hashable type, such as a tuple, to ensure that the key is always hashable.